src/memory_service/fam_memory_service_direct.cpp

```diff
@@ -119,7 +119,12 @@ Fam_Memory_Service_Direct::Fam_Memory_Service_Direct(
         atoi(config_options["delayed_free_threads"].c_str());
 
     allocator = new Memserver_Allocator(num_delayed_free_Threads, fam_path);
-
+    fam_backup_path = config_options["fam_backup_path"];
+    struct stat info;
+    if (stat(fam_backup_path.c_str(), &info) == -1) {
+        if ((errno == ENOENT) || (errno == ENOTDIR))
+            mkdir(fam_backup_path.c_str(), 0777);
```

Comment:
Let us not explicitly set permissions here, it will take permissions from umask and default permission. Since memory server will be run using same user it should be fine.